internal/services/inference/deployment.go

```diff
@@ -0,0 +1,308 @@
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The model name to use for the deployment",
+			},
+			"model_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The model id used for the deployment",
+			},
+			"accept_eula": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Whether or not the deployment is accepting eula",
+			},
+			"tags": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "The tags associated with the deployment",
+			},
+			"min_size": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "The minimum size of the pool",
+			},
+			"max_size": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "The maximum size of the pool",
+			},
+			"size": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The size of the pool",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the deployment",
+			},
+			"endpoint_public_url": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The endpoint public URL",
+			},
+			"endpoint_private_url": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The endpoint private URL",
+			},
+			"disable_auth_private": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Whether or not the authentication on the private endpoint is disabled.",
+			},
+			"endpoint_public_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The endpoint public ID",
+			},
+			"endpoint_private_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The endpoint private ID",
+			},
+			"disable_auth_public": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Whether or not the authentication on the public endpoint is disabled.",
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the creation of the deployment",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the last update of the deployment",
+			},
+			"endpoints": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "List of endpoints",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"public_endpoint": {
+							Type:        schema.TypeBool,
+							Description: "Set the endpoint as public",
+							Optional:    true,
+						},
+						"private_endpoint": {
+							Type:        schema.TypeString,
+							Description: "The id of the private network",
+							Optional:    true,
+						},
```

Comment:
Having `public_endpoint` as a bool and `private_endpoint` as an ID is misleading.
These fields should have names that suit their use, example: `is_public`, `private_network_id`